### ISSUE TITLE
Bug after upgrade | sdwan_hub_and_spoke_topology_policy_definition

### ISSUE THREAD

**Issue by rrahimm** [CiscoDevNet/terraform-provider-sdwan#230]
After updating the provider to latest, getting the below error for sdwan_hub_and_spoke_topology_policy_definition

╷
│ Error: Provider returned invalid result object after apply
│ 
│ After the apply operation, the provider still indicated an unknown value for
│ module.sdwan.sdwan_hub_and_spoke_topology_policy_definition.hub_and_spoke_topology_policy_definition["HST_DEFINITION_T1-rrahimm"].type. All
│ values must be known after apply, so this is always a bug in the provider and should be reported in the provider's own repository. Terraform
│ will still save the other known object values in the state.

Running version

<img width="285" alt="image" src="https://github.com/CiscoDevNet/terraform-provider-sdwan/assets/39684933/e8e054b4-b917-40c2-8c69-2956cd960476">

**Comment by rrahimm** [comment]
Happening when using the below commented fields, that were added as part of https://github.com/CiscoDevNet/terraform-provider-sdwan/issues/112

   ```
     hub_and_spoke_topology:
          - name: HST_DEFINITION_T1-rrahimm
            description: HST_DEFINITION_T1-rrahimm
            vpn_list: VPN-LIST-HST-T1-rrahimm
            hub_and_spoke_sites:
              - name: group1
                spokes:
                  - site_list: CALIFORNIA-HST-T-rrahimm
                    hubs:
                      - site_list: BOSTON-HST-T-rrahimm
                      - site_list: NEWYORK-HST-T-rrahimm
                        # ipv4_prefix_lists:
                        #   - PREFIX-LIST-HST-T-rrahimm
              - name: group2
                equal_preference: true
                # advertise_tloc: true
                # tloc_list: TLOC-LIST-HST-T2-rrahimm
                spokes:
                  - site_list: WASHINGTON-HST-T-rrahimm
                    hubs:
                      - site_list: KENTUCKY-HST-T-rrahimm
          - name: HST_DEFINITION_T2-rrahimm
            description: HST_DEFINITION_T2-rrahimm
            vpn_list: VPN-LIST-HST-T1-rrahimm
            hub_and_spoke_sites:
              - name: group1
                equal_preference: true
                # advertise_tloc: true
                # tloc_list: TLOC-LIST-HST-T1-rrahimm
                spokes:
                  - site_list: DENVER-CCT-T-rrahimm
                    hubs:
                      - site_list: ATLANTA-CCT-T-rrahimm
                        # ipv4_prefix_lists:
                        #   - PREFIX-LIST-HST-T-rrahimm
```

**Comment by danischm** [comment]
I can't seem to recreate it based on what you shared. Could you maybe share a sample resource definition to recreate the issue?

**Comment by rrahimm** [comment]
Hello @danischm 

i think, even i am not able to recreate if I push all parameters at once.

what I am doing is.

Firstly, create the policy without the parameters advertise_hub_tlocs, all_hubs_are_equal and tloc_list_id.

Then I try to add these 3 variables to the policy and encounter error.

Both parts of the policy were created by the newer Provider version.


```
Terraform will perform the following actions:

  # module.sdwan.sdwan_hub_and_spoke_topology_policy_definition.hub_and_spoke_topology_policy_definition["HST_DEFINITION_T3-rrahimm"] will be created
  + resource "sdwan_hub_and_spoke_topology_policy_definition" "hub_and_spoke_topology_policy_definition" {
      + description      = "HST_DEFINITION_T3-rrahimm"
      + id               = (known after apply)
      + name             = "HST_DEFINITION_T3-rrahimm"
      + topologies       = [
          + {
              + name   = "group1"
              + spokes = [
                  + {
                      + hubs              = [
                          + {
                              + site_list_id      = "6b18c5c7-45a6-4b59-9f77-e2e9fcc4ae64"
                              + site_list_version = 0
                            },
                        ]
                      + site_list_id      = "8a84db66-2aba-462e-bfa8-ff9e9924515b"
                      + site_list_version = 0
                    },
                ]
            },
        ]
      + type             = (known after apply)
      + version          = (known after apply)
      + vpn_list_id      = "5e94d34c-8c88-4a49-980b-89265ba4e6ae"
      + vpn_list_version = 0
    }
    
    
    
    erraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # module.sdwan.sdwan_hub_and_spoke_topology_policy_definition.hub_and_spoke_topology_policy_definition["HST_DEFINITION_T3-rrahimm"] will be updated in-place
  ~ resource "sdwan_hub_and_spoke_topology_policy_definition" "hub_and_spoke_topology_policy_definition" {
        id               = "615e5395-aa22-44c5-8093-3d3c8e58d075"
        name             = "HST_DEFINITION_T3-rrahimm"
      ~ topologies       = [
          ~ {
              + advertise_hub_tlocs = true
              + all_hubs_are_equal  = true
                name                = "group1"
              + tloc_list_id        = "e5439673-0973-40d5-a378-165ad63d35af"
                # (1 unchanged attribute hidden)
            },
        ]
      ~ type             = "hubAndSpoke" -> (known after apply)
      ~ version          = 0 -> (known after apply)
        # (3 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
module.sdwan.sdwan_hub_and_spoke_topology_policy_definition.hub_and_spoke_topology_policy_definition["HST_DEFINITION_T3-rrahimm"]: Modifying... [id=615e5395-aa22-44c5-8093-3d3c8e58d075]
╷
│ Error: Provider returned invalid result object after apply
│ 
│ After the apply operation, the provider still indicated an unknown value for
│ module.sdwan.sdwan_hub_and_spoke_topology_policy_definition.hub_and_spoke_topology_policy_definition["HST_DEFINITION_T3-rrahimm"].type. All
│ values must be known after apply, so this is always a bug in the provider and should be reported in the provider's own repository. Terraform
│ will still save the other known object values in the state.
```

**Comment by rrahimm** [comment]
Terraform Module

```
resource "sdwan_hub_and_spoke_topology_policy_definition" "hub_and_spoke_topology_policy_definition" {
  for_each         = { for d in try(local.centralized_policies.definitions.control_policy.hub_and_spoke_topology, {}) : d.name => d }
  name             = each.value.name
  description      = each.value.description
  vpn_list_id      = sdwan_vpn_list_policy_object.vpn_list_policy_object[each.value.vpn_list].id
  vpn_list_version = sdwan_vpn_list_policy_object.vpn_list_policy_object[each.value.vpn_list].version
  topologies = try(length(each.value.hub_and_spoke_sites) == 0, true) ? null : [for t in each.value.hub_and_spoke_sites : {
    name = t.name
    all_hubs_are_equal = try(t.equal_preference, null)
    advertise_hub_tlocs = try(t.advertise_tloc, null)
    tloc_list_id = try(t.tloc_list, null) == null ? null : sdwan_tloc_list_policy_object.tloc_list_policy_object[t.tloc_list].id
    spokes = try(length(t.spokes) == 0, true) ? null : [for s in t.spokes : {
      site_list_id      = sdwan_site_list_policy_object.site_list_policy_object[s.site_list].id
      site_list_version = sdwan_site_list_policy_object.site_list_policy_object[s.site_list].version
      hubs = try(length(s.hubs) == 0, true) ? null : [for h in s.hubs : {
        site_list_id      = sdwan_site_list_policy_object.site_list_policy_object[h.site_list].id
        site_list_version = sdwan_site_list_policy_object.site_list_policy_object[h.site_list].version
        ipv4_prefix_list_ids = try(length(h.ipv4_prefix_lists) == 0, true) ? null : [for p in h.ipv4_prefix_lists : sdwan_ipv4_prefix_list_policy_object.ipv4_prefix_list_policy_object[p].id]
        ipv6_prefix_list_ids = try(length(h.ipv6_prefix_lists) == 0, true) ? null : [for p in h.ipv6_prefix_lists : sdwan_ipv6_prefix_list_policy_object.ipv6_prefix_list_policy_object[p].id]
        }
      ]
    }]
  }]
}
```

**Comment by danischm** [comment]
Should be fixed here: https://github.com/CiscoDevNet/terraform-provider-sdwan/commit/53ffe4690211dec76c4ab4932ef19db43950df32

**Comment by danischm** [comment]
Published in v0.3.9 release.

**Comment by rrahimm** [comment]
Thanks for the quick fix @danischm
